### PR TITLE
[Issue #3291] Create database tables for saved opportunities

### LINKED ISSUE #3291: Create database tables for saved opportunities
### Summary

We want to create a table named `user_saved_opportunity` with the following columns:

* user_id - UUID (Primary Key)
* opportunity_id - int (Primary Key)
* created_at/updated_at (as usual from the mixin)

### Acceptance criteria
- [ ] Table created
- [ ] Migrations created

### PR DESCRIPTION
**Pull request by mikehgrantsgov**
## Summary
Fixes #3291

## Changes proposed
Add a `user_saved_opportunity` table with columns as described in the ticket.

## Context for reviewers
Needed for follow-on work with saved opportunity API tickets.

### CODE REVIEW

File: api/src/db/models/user_models.py [hunk 1]
```diff
@@ -60,3 +60,12 @@ class LoginGovState(ApiSchemaTable, TimestampMixin):
 
     # https://openid.net/specs/openid-connect-core-1_0.html#NonceNotes
     nonce: Mapped[uuid.UUID]
+
+
+class UserSavedOpportunity(ApiSchemaTable, TimestampMixin):
+    __tablename__ = "user_saved_opportunity"
+
+    user_id: Mapped[uuid.UUID] = mapped_column(ForeignKey(User.user_id), primary_key=True)
+    opportunity_id: Mapped[int] = mapped_column(primary_key=True)
```
Comment (chouinar) [inline api/src/db/models/user_models.py:69]: We would want this one to be a foreign key as well to the opportunity table

Comment (mikehgrantsgov) [reply api/src/db/models/user_models.py]: Ah right, added that.

File: api/src/db/models/user_models.py [hunk 2]
```diff
@@ -60,3 +61,14 @@ class LoginGovState(ApiSchemaTable, TimestampMixin):
...
     nonce: Mapped[uuid.UUID]
+
+
+class UserSavedOpportunity(ApiSchemaTable, TimestampMixin):
+    __tablename__ = "user_saved_opportunity"
+
+    user_id: Mapped[uuid.UUID] = mapped_column(ForeignKey(User.user_id), primary_key=True)
+    opportunity_id: Mapped[int] = mapped_column(
+        ForeignKey(Opportunity.opportunity_id), primary_key=True
+    )
```
Comment (chouinar) [inline api/src/db/models/user_models.py:72]: Oh, right we'd want this to be a BigInteger so that the type matches opportunity ID exactly

Comment (mikehgrantsgov) [reply api/src/db/models/user_models.py]: Updated this!

File: api/src/db/models/user_models.py [hunk 3]
```diff
@@ -60,3 +61,14 @@ class LoginGovState(ApiSchemaTable, TimestampMixin):
...
     nonce: Mapped[uuid.UUID]
+
+
+class UserSavedOpportunity(ApiSchemaTable, TimestampMixin):
+    __tablename__ = "user_saved_opportunity"
+
+    user_id: Mapped[uuid.UUID] = mapped_column(ForeignKey(User.user_id), primary_key=True)
+    opportunity_id: Mapped[BigInteger] = mapped_column(
+        ForeignKey(Opportunity.opportunity_id), primary_key=True
+    )
```
Comment (chouinar) [inline api/src/db/models/user_models.py:72]: Sorry to keep pestering on this, I'd want this so the python typing is "right" as well.
```suggestion
    opportunity_id: Mapped[int] = mapped_column(
        BigInteger, ForeignKey(Opportunity.opportunity_id), primary_key=True
    )
```

Comment (mikehgrantsgov) [reply api/src/db/models/user_models.py]: Oh you're right, just had the wrong type there.

**Comment by mikehgrantsgov** [comment]
I think we are good now with the typing stuff.

**Review by chouinar — APPROVED** [review]

**Pull request merged**